### DISCUSSION TITLE
Tailwind with PostCSS: JIT mode no longer picks up on changes to content files

### DISCUSSION THREAD

**Discussion by nikitaourazbaev** [tailwindlabs/tailwindcss#7045] [Help] [answered]
Not sure if this should be an issue.

Here’s a reduced test case:

https://github.com/nikitaourazbaev/tw-issue-repro

After updating to Tailwind 3, PostCSS / Tailwind no longer picks up on changes in content (HTML) files until I touch either `tailwind.config.js` or the source css file. This setup worked fine with Tailwind v2 in JIT mode and this setup also works now with Tailwind CLI but running it this way, with Tailwind as a PostCSS plugin, watching doesn’t work.

I’ve tried removing `TAILWIND_MODE=watch` but it does not seem to have an effect.

Am I doing something wrong with the setup here?

Steps to reproduce:

Run the `install` script, then the `dev` script. Then change the `bg-amber-200 text-amber-500` classes on the inner div in `index.html` to any other colors.

The expected result is that the CSS recompiles, as it does with Tailwind v2 and with the Tailwind CLI. Instead, nothing happens until I run the `dev` script again or touch either the tailwind config file or the CSS file.

**Comment by paragchirde** [comment]
I am facing the same issue. I am currently running the project on M1 pro mac with Node v16 and Tailwind v3. Tried creating the `.env `file as well as adding `TAILWIND_MODE=watch` in my dev script. Both don't seem to be working. 
I raised the issue in the same as well #7020

**Comment by nikitaourazbaev** [comment, marked as answer]
The solution in my case was to update `postcss-cli` to `^9`, as per https://github.com/tailwindlabs/tailwindcss/issues/7136
